etc/clang_plugin/plugin_topfname.cpp

```diff
@@ -175,16 +177,57 @@ namespace llvm
       }
 #endif
 
-      bool exec(Module& M)
+      bool exec(Module& M, const CallGraph& CG)
       {
          bool changed = false;
          bool hasTopFun = false;
+         std::list<std::string> symbolList;
+         std::vector<std::string> Starting_TopFunctionNames;
```

Comment:
It is not clear why this is necessary. It is just a copy of TopFunctionNames.

Reply:
Because if you keep one list, you cannot prune the functions from architecture.xml that are not reachable from the TopFunctionName_TFP.